js/scenes/FoodScreen/FoodScreen.js

```diff
@@ -34,14 +45,24 @@ const FoodScreen = ({ userId, userName, potluckFood, isLoading }) => (
       {potluckFood.food && Object.keys(potluckFood.food).map(key => (
         <FoodSelection
           key={key}
-          username={potluckFood.food[key].username || ''}
-          count={potluckFood.food[key].desiredDishCount}
           foodItem={POTLUCK_FOOD[key]}
+          availableFoodItemCount={potluckFood.food[key].desiredDishCount -
+            (doesObjectPropertyExist(potluckFood.food[key], 'assignments') ?
+              Object.keys(potluckFood.food[key].assignments).length :
+              0
+            )
+          }
```

Comment:
Could you generalize that reducer function? It looks like it's doing the same thing three times.